### PR TITLE
[Depatures][Rennes] Clean memory cache at each new processing day

### PR DESCRIPTION
**Pull request by julienbt**

### CODE REVIEW

File: internal/departures/rennes/stoptimes/stoptimes.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ import (
 
 const StopTimesFileName string = "horaires.hor"
 const stopTimesCsvNumOfFields int = 4
+const ProcessingEndTimeHour int = 1
+const ProcessingEndTimeMinute int = 30
```
Comment (kadhikari) [inline internal/departures/rennes/stoptimes/stoptimes.go:16]: These two variables are used ?

File: internal/departures/rennes/rennes.go [hunk 2]
```diff
@@ -107,21 +177,49 @@ func (d *RennesContext) InitContext(
...
+	logrus.Infof("The depatures list is reset, the daily service start time is: %v", dailyServiceStartTime)
 }
 
 func (d *RennesContext) RefereshDeparturesLoop(context *departures.DeparturesContext) {
 	context.SetPackageName(reflect.TypeOf(RennesContext{}).PkgPath())
 	context.SetFilesRefeshTime(d.connector.GetFilesRefreshTime())
 	context.SetWsRefeshTime(d.connector.GetWsRefreshTime())
 
-	// Wait 10 seconds before reloading external departures informations
-	time.Sleep(2 * time.Second)
```
Comment (kadhikari) [inline internal/departures/rennes/rennes.go:121]: You can leave it here

**Review by kadhikari — CHANGES_REQUESTED** [review]
Simplify some codes on date time management.
Some examples in forseti:
* https://github.com/hove-io/forseti/blob/dc576e3a9c35f23c6913ccc81b214d4b5acd3429/internal/vehicleoccupancies/navitia.go#L187
* https://github.com/hove-io/forseti/blob/dc576e3a9c35f23c6913ccc81b214d4b5acd3429/internal/vehicleoccupancies/navitia.go#L111
* https://github.com/hove-io/forseti/blob/f95e4d3f589746c756ea61d4a1c4e32866cc5b31/internal/vehiclepositions/contextGtfsRt.go#L162

File: internal/departures/rennes/rennes.go (hunk 2 above)
Comment (julienbt) [reply internal/departures/rennes/rennes.go:121]: Done

File: cmd/forseti/main.go [hunk 3]
```diff
@@ -226,7 +229,19 @@ func main() {
...
+	{
+		t, _ := time.ParseInLocation(
+			"15:04:05", config.DeparturesServiceSwitch, location)
+		departuresServiceSwitchTime = time.Date(
+			0, 1, 1,
+			t.Hour(), t.Minute(), t.Second(), t.Nanosecond(),
+			location,
+		)
+	}
+	fmt.Printf("serviceSwitchTime: %v\n", departuresServiceSwitchTime)
```
[the remark this answers is not in the document]
Comment (julienbt) [reply cmd/forseti/main.go:243]: Removed

File: internal/departures/rennes/stoptimes/stoptimes_test.go [hunk 4]
```diff
@@ -116,3 +116,129 @@ func TestLoadStopTimes(t *testing.T) {
...
+		// {
+		// 	name: "0 nanosecond after service start time (midnight)",
+		// 	stopTime: StopTime{
+		// 		Id:               "UNUSED_ID",
+		// 		Time:             time.Date(0, time.January, 1, 0, 0, 0, 0, EUROPE_PARIS_LOCATION),
+		// 		RouteStopPointId: "UNUSED_ROUTE_STOP_POINT_ID",
+		// 	},
+		// 	dailyServiceStartTime:  SERVICE_START_TIME_AT_MIDNIGHT,
+		// 	expectedActualStopTime: time.Date(2022, time.June, 15, 0, 0, 0, 0, EUROPE_PARIS_LOCATION),
+		// },
```
Comment (kadhikari) [inline internal/departures/rennes/stoptimes/stoptimes_test.go:216]: Can you delete the commented block ?

**Review by kadhikari — APPROVED** [review]

**Pull request merged**